### PR TITLE
French: added translated color for french language

### PR DESCRIPTION
**Pull request by RSickenberg**
### What is the reason for this PR?

- [x] A new feature
- [ ] Fixed an issue (resolve #ID)

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- [ ] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)

### Summary of changes

I simply added a list of colors we use in french.

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

File: src/Faker/Calculator/Isbn.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ public static function checksum(string $input): string
         $digits = str_split($input);
         array_walk(
             $digits,
-            static function (&$digit, $position) {
+            static function (& $digit, $position) {
```
Comment (RSickenberg) [inline src/Faker/Calculator/Isbn.php:37]:
```suggestion
            static function (&$digit, $position) {
```

File: src/Faker/UniqueGenerator.php [hunk 2]
```diff
@@ -33,7 +33,7 @@ class UniqueGenerator
      * @param int                                $maxRetries
      * @param array<string, array<string, null>> $uniques
      */
-    public function __construct($generator, $maxRetries = 10000, &$uniques = [])
+    public function __construct($generator, $maxRetries = 10000, & $uniques = [])
```
Comment (RSickenberg) [inline src/Faker/UniqueGenerator.php:36]:
```suggestion
    public function __construct($generator, $maxRetries = 10000, &$uniques = [])
```

**Comment by pimjansen** [comment]
Cant these classes all extend from FR_fr to remove duplication?

**Comment by RSickenberg** [comment]
Good point @pimjansen will do!

**Comment by RSickenberg** [comment]
@pimjansen Let me know if it's ok for you. I used `fr_CH` by default because it's my IRL locale 😁 but I can change it.

**Comment by pimjansen** [comment]
> @pimjansen Let me know if it's ok for you. I used `fr_CH` by default because it's my IRL locale 😁 but I can change it. 

Lets use FR_fr instead since its the base locale for it

**Comment by RSickenberg** [comment]
@pimjansen Done!

File: src/Faker/Calculator/Isbn.php (hunk 1 above)
Comment (pimjansen) [inline src/Faker/Calculator/Isbn.php:37]: How is this relevant to the PR? Please revert

File: src/Faker/UniqueGenerator.php (hunk 2 above)
Comment (pimjansen) [inline src/Faker/UniqueGenerator.php:36]: Not relevant to this PR, please revert

**Review by pimjansen — CHANGES_REQUESTED** [review]

**Comment by pimjansen** [comment]
PR looks ok besides the two mentioned topics

File: src/Faker/Calculator/Isbn.php (hunk 1 above)
Comment (RSickenberg) [reply src/Faker/Calculator/Isbn.php:37]: This was due to the CS script.

File: src/Faker/UniqueGenerator.php (hunk 2 above)
Comment (RSickenberg) [inline src/Faker/UniqueGenerator.php:36]:
```suggestion
    public function __construct($generator, $maxRetries = 10000, &$uniques = [])
```

**Comment by RSickenberg** [comment]
@pimjansen Done!

Please, note it was due to the CS linter ran with PHP 8.1.4, it sure is off topic but you may like to checks this 😄

File: src/Faker/Calculator/Isbn.php (hunk 1 above)
Comment (pimjansen) [reply src/Faker/Calculator/Isbn.php]: Which one?

**Comment by pimjansen** [comment]
One last thing i did not mentioned earlier (sorry for that though its in the guide). Please ensure the new code is covered by unittests

**Comment by RSickenberg** [comment]
@pimjansen Do you want me to test an array? That's not even testable 🥲. First time I truly contribute to an open-source project, now I understand why most devs won't do this in their career. (sorry for that though, that's my feeling)

**Comment by pimjansen** [comment]
Sorry that you feel this way! But as mentioned in the contribution guide which you manually checked it states:

* Your code should be covered by unit tests.

Since this is mandatory in libraries used all over the place. The test now actually is testing is something is a string which actually does not test much. Faker works with seeding so please assert an actual value from the seed so we can ensure this does not change. You might think that you are just testing an array value but there could actually be quite some logic behind it.

This test is both for the safeColor as allColor methods from the Faker Generator

**Comment by RSickenberg** [comment]
@pimjansen That's ok, it's more a question of "how" than "why/what". 

Can you provide me an example (or something similar) where I can see how to seed?

**Comment by bram-pkg** [comment]
Seeding is done automatically in the base test class. Look at this test for example: https://github.com/FakerPHP/Faker/blob/main/test/Faker/Core/DateTimeTest.php

**Comment by RSickenberg** [comment]
So, @bram-pkg my tests are ok?

**Comment by bram-pkg** [comment]
Not quite, make sure to compare them against the output of your function. Since it's being seeded, it will always be the same.

**Comment by RSickenberg** [comment]
Don't. I will find time to write those tests

**Comment by RSickenberg** [comment]
Honestly @bram-pkg and @pimjansen, I don't quite understand. I didn't even write some functions in this PR. I only translated some strings. 

Can you pin me exactly what did you expect? The `DateTimeTest` you sent me is not relevant since I don't know what to look at.

**Comment by pimjansen** [comment]
> Honestly @bram-pkg and @pimjansen, I don't quite understand. I didn't even write some functions in this PR. I only translated some strings. 
> 
> 
> 
> Can you pin me exactly what did you expect? The `DateTimeTest` you sent me is not relevant since I don't know what to look at.

We are changing the dataset so we want to ensure it stays intact and does what expected. You now added some regex to see if its a valid string which on its own is fine.

Since Faker is using seeding it means that a testcase always returns the same result.

Therefor i suggest to test an actual value of the generator. In your case assert that the generator returns the value which is returned while calling it with a specific seed.

**Comment by pimjansen** [comment]
@RSickenberg can you allow us to make changes to your PR to finish it?

**Comment by RSickenberg** [comment]
> @RSickenberg can you allow us to make changes to your PR to finish it?

Sure, please do. I don't have time to contribute yet 😔

**Comment by RSickenberg** [comment]
> > > @RSickenberg can you allow us to make changes to your PR to finish it?
> > 
> > 
> > @pimjansen Sure, please do. I don't have time to contribute yet 😔
> 
> You have to allow this since i cant at this moment

image
It's already on 🤔

File: src/Faker/Provider/fr_FR/Color.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Faker\Provider\fr_FR;
+
+class Color extends \Faker\Provider\Color
+{
+    protected static $safeColorNames = [
+        'noir', 'maron', 'vert', 'marine', 'olive',
```
Comment (fgaroby) [inline src/Faker/Provider/fr_FR/Color.php:8]: 'marron' (avec 2 'n')

Comment (RSickenberg) [reply src/Faker/Provider/fr_FR/Color.php:8]: Bien vu!

**Comment by RSickenberg** [comment]
@pimjansen Any updates ?

**Review by pimjansen — APPROVED** [review]

**Comment by RSickenberg** [comment]
Thank you @pimjansen ❤️.

**Comment by pimjansen** [comment]
> Thank you @pimjansen ❤️. 

Is it clear for you how the seeding works now and the diff i did in the test?

**Comment by RSickenberg** [comment]
@pimjansen Yup, not gonna lie, I thought the call was about to return random choices instead of the one asserted.

**Comment by pimjansen** [comment]
> @pimjansen Yup, not gonna lie, I thought the call was about to return random choices instead of the one asserted.

No worries, now you know for the future 😋
